lib/MessageBatch.cc

```diff
@@ -47,7 +47,7 @@ MessageBatch& MessageBatch::parseFrom(const SharedBuffer& payload, uint32_t batc
     batch_.clear();
 
     for (int i = 0; i < batchSize; ++i) {
-        batch_.push_back(Commands::deSerializeSingleMessageInBatch(batchMessage_, i));
+        batch_.push_back(Commands::deSerializeSingleMessageInBatch(batchMessage_, i, batchSize));
```

Comment:
Have a question: What is the role of `MessageBatch`, Doesn't seem to find a place to use it.

Reply:
https://github.com/apache/pulsar/pull/4992

Though it's not a good design IMO.

Reply:
Get it, Thanks.